### PR TITLE
feat: support default configuration for client objects

### PR DESCRIPTION
**Pull request by cprice404**
Based on some feedback about example code and related content,
this commit makes the Configuration arguments optional for all
Client constructors, and provides default configurations for
each one. This will allow us to simplify our code examples
so that it is easier to highlight the more interesting and
important parts of the code.

### CODE REVIEW

File: packages/client-sdk-nodejs/src/config/vector-index-configurations.ts [hunk 1]
```diff
@@ -61,47 +61,3 @@ export class Laptop extends VectorIndexClientConfiguration {
     });
   }
 }
-
-/**
- * Browser config provides defaults suitable for use in a web browser.
- * @export
- * @class Browser
- */
-export class Browser extends VectorIndexClientConfiguration {
```
Comment (malandis) [inline packages/client-sdk-nodejs/src/config/vector-index-configurations.ts:70]: Nice catch

File: packages/client-sdk-nodejs/src/internal/cache-client-props-with-config.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import {CacheClientProps} from '../cache-client-props';
+import {Configuration} from '../config/configuration';
+
+export interface CacheClientPropsWithConfig extends CacheClientProps {
+  configuration: Configuration;
+}
```
Comment (malandis) [inline packages/client-sdk-nodejs/src/internal/cache-client-props-with-config.ts:6]: I wasn't aware you could override a property's optionality (`configuration` is optional now in `CacheClientProps`) this way. That's convenient.

**Review by malandis — APPROVED** [review]
👍

**Review by bruuuuuuuce — APPROVED** [review]
:shipit:

**Pull request merged**